apluslms_roman/utils/path_mapping.py

```diff
@@ -0,0 +1,30 @@
+from pathlib import PurePosixPath
+import json
+
+
+def get_host_path(original, mapping):
+    ret = original
+    orig_path = PurePosixPath(original)
+    for k, v in mapping.items():
+        try:
+            print(k, v)
+            relative_path = orig_path.relative_to(k)
+            ret = PurePosixPath(v).joinpath(relative_path)
+            return str(ret)
+        except ValueError:
+            print("Error when composing new path!")
+            return -1
+    return str(ret)
+
+
+def get_pair_form_env(key, json_str):
```

Comment:
This function could be refactored to something called `def read_from_environ(prefix: str) -> dict:`. It would:

1. read all values from `os.environ`, where key starts with `prefix`
2. checks if the value matches re `r'^(?:["[{]|(?:\d+|null|true|false)$)', if so then json decode it
3. if decode fails, raise error
4. lowercase the key
5. if there is `.` (dot) in the key, then split the key to parts (e.g. `"FOO_BAR.BAZ"` -> `["FOO_BAR", "BAZ"]`)
6. build result nested dictionaries from the data, presuming that missing elements are dicts

For example:
```
os.environ = {
  'PREFIX_FOO.BAR': 'true',
  'PREFIX_FOO.BAZ': '[1, 2, 3]',
}
read_from_environ('PREFIX_') == {
  'foo': {
    'bar': True,
    'baz': [1, 2, 3],
  }
}
```

Reply:
>     5\. if there is `.` (dot) in the key, then split the key to parts (e.g. `"FOO_BAR.BAZ"` -> `["FOO_BAR", "BAZ"]`)

You mean in this case we have two keys now?

Reply:
In such case the setting cannot overwrite the environment variable, in such case, the environment variable after reading setting is :
```
{'directory_map': {'foo': 'bar', 'foo1': 'bar1'},
 'DOCKER_DIRECTORY_MAP': ChangesDict(data={'/data/': '/var/lib/docker/volumes/aplus_data/_data/'}, default={})}
```
Which shows that two variable exist at same time, see slack chat for more deatil